09_dynamic_programming/csharp/02_levenshtein/Program.cs

```diff
@@ -0,0 +1,53 @@
+using System;
+
+namespace ConsoleApplication
+{
+    public class Program
+    {
+        public static void Main(string[] args)
+        {
+            var result = LevenshteinDistance("gibraltar", "labrador");
+            Console.WriteLine(result); // 5
+        }
+
+        public static int LevenshteinDistance(string source, string target)
+        {
+            var matrix = CreateMatrix(source, target);
+
+            for (int i = 1; i <= source.Length; i++)
+            {
+                for (int j = 1; j <= target.Length; j++)
+                {
+                    matrix[i, j] = Math.Min(matrix[i, j - 1] + 1, Math.Min(
+                        matrix[i - 1, j] + 1, matrix[i - 1, j - 1] + (source[i - 1] != target[j - 1] ? 1 : 0)));
+                }
+            }
+
+            return matrix[source.Length, target.Length];
+        }
+
+        private static int[,] CreateMatrix(string source, string target)
+        {
+            var matrix = new int[source.Length + 1, target.Length + 1];
+
+            if (source.Length < target.Length)
+            {
+                var tempSrc = source;
```

Comment:
You don't need a temporary var I believe https://stackoverflow.com/a/39190792/20488510